src/wallet/api/wallet.h

```diff
@@ -151,6 +152,10 @@ class WalletImpl : public Wallet
     bool                m_recoveringFromSeed;
     std::atomic<bool>   m_synchronized;
     bool                m_rebuildWalletCache;
+    // cache connection status to avoid unnecessary RPC calls
+    mutable bool        m_is_connected;
+    // on asynchronous wallet close we need to make sure no calls are made after shutdown started.
+    bool                m_is_shutting_down;
```

Comment:
Does this need to be atomic? It looks like the flag is written to and read from by different threads, which means there is no guarantee that the read has the most up-to-date value. I think the issue is minor (ignoring the purists).